src/gui/folder.cpp

```diff
@@ -763,7 +695,7 @@ void Folder::slotWatchedPathsChanged(const QSet<QString> &paths, ChangeReason re
 
         // Also schedule this folder for a sync, but only after some delay:
         // The sync will not upload files that were changed too recently.
-        scheduleThisFolderSoon();
+        QTimer::singleShot(SyncEngine::minimumFileAgeForUpload, this, [this] { FolderMan::instance()->scheduler()->enqueueFolder(this); });
```

Comment:
Could these single shot timers accumulate over time? We recently saw an issue where Windows's timer limit was reached.

Reply:
its a 3ms timer, which should not exist longer than 3ms